waspc/src/Wasp/Generator/FileDraft/CopyAndModifyTextFileDraft.hs

```diff
@@ -0,0 +1,54 @@
+module Wasp.Generator.FileDraft.CopyAndModifyTextFileDraft
+  ( CopyAndModifyTextFileDraft (..),
+  )
+where
+
+import Control.Monad (when)
+import qualified Data.Text as T
+import qualified Data.Text.IO as T.IO
+import StrongPath (Abs, File', Path', Rel, (</>))
+import qualified StrongPath as SP
+import System.IO.Error (doesNotExistErrorType, mkIOError)
+import Wasp.Generator.Common (ProjectRootDir)
+import Wasp.Generator.FileDraft.Writeable
+import Wasp.Generator.FileDraft.WriteableMonad
+import Wasp.Util (checksumFromText)
+
+-- | File draft based on existing text file + some modifications.
+data CopyAndModifyTextFileDraft = CopyAndModifyTextFileDraft
+  { -- | Path where the file will be copied to.
+    _dstPath :: !(Path' (Rel ProjectRootDir) File'),
+    -- | Absolute path of source file to copy.
+    _srcPath :: !(Path' Abs File'),
+    _failIfSrcDoesNotExist :: !Bool,
+    _modify :: !(T.Text -> T.Text)
+  }
+
+instance Writeable CopyAndModifyTextFileDraft where
```

Comment:
This is neat! :D Nice

Reply:
Yeah I was not sure if I should do it or not to be honest!
Name turned out really long.
I was tempted to instead modify `CopyFileDraft` and make it have additional field that allows this modification.
But then that felt like I will be using too power full(draft) in 90% of cases so I decided to make this separate one.
Hm.

Reply:
Actually I ended up not using this at the end, but I am considering leaving it in due to amount of work + potential it might be useful in the future. If you all think this is wrong decision let me know and I can remove it still.

Reply:
If we aren't using it, I wonder if it can easily be extracted into some other branch for future possible use? Only downside to keeping it if not used is the additional drag anytime we make `FileDraft` changes, we have another thing to update.

Reply:
Yeah I know. Ah man. Probably best to remove it then!